plugins/scanners/fossid/src/main/kotlin/FossId.kt

```diff
@@ -922,4 +927,43 @@ class FossId internal constructor(
             mapOf(SCAN_CODE_KEY to scanCode, SCAN_ID_KEY to scanId, SERVER_URL_KEY to config.serverUrl)
         )
     }
+
+    /**
+     * Mark all the files having a snippet choice as identified, only if they have no non-chosen source location
+     * remaining.
+     */
+    private fun markFilesWithChosenSnippetsAsIdentified(
+        scanCode: String,
+        snippetChoices: List<SnippetChoice> = emptyList(),
+        snippetFinding: Set<SnippetFinding>,
+        listPendingFiles: List<String>
+    ) {
+        runBlocking(Dispatchers.IO) {
+            val candidatePathsToMark = snippetChoices.groupBy({ it.given.sourceLocation.path }) {
+                it.choice.reason
+            }
+
+            val requests = mutableListOf<Deferred<Any>>()
+
+            candidatePathsToMark.forEach { (path, reasons) ->
+                val allLocationsChosen = snippetFinding.none { path == it.sourceLocation.path }
+
+                if (allLocationsChosen) {
+                    if (listPendingFiles.none { path == it }) {
+                        logger.info { "Not marking $path as it it not pending." }
+                    } else {
+                        logger.info {
+                            "Marking $path as identified as it has the following snippet choices $reasons"
```

Comment:
I'm not sure if I understand that message, how about that?
`"Marking $path as identified as for all snippets a choice has been made. The used reasons are: $[reasons.joinToString()}"`

Reply:
I adapted the sentence a bit further.